File: cse-308/offline-2/problem-2/src/Student.cpp
#include "Student.h"

Student::Student(Controller *controller)
{
    this->controller = controller;

    SetID(0);
    SetMarks(101);
}

Student::Student(size_t id, Controller *controller)
{
    this->controller = controller;

    SetID(id);
    SetMarks(101);
}

void Student::SetID(size_t id)
{
    this->id = id;
}

void Student::SetMarks(size_t marks)
{
    this->marks = marks;
}

size_t Student::GetID() const
{
    return id;
}

size_t Student::GetMarks() const
{
    return marks;
}

void Student::Appeal(size_t claim)
{
    controller->Appeal(this, claim);
}

Student::~Student()
{

}

File: cse-308/offline-2/problem-2/src/problem2/Student.java
package problem2;

public class Student extends AcademicComponent
{
    private int id;
    private int marks;

    public Student(int id, Controller controller)
    {
        super(controller);
        SetMarks(101); // 101 means not set yet, kinda null
    }

    public void Notify(String command, String params)
    {
        controller.Notify(this, command, params);
    }

    public void SetID(int id)
    {
        this.id = id;
    }

    public int GetID()
    {
        return id;
    }

    public void SetMarks(int marks)
    {
        this.marks = marks;
    }

    public int GetMarks()
    {
        return marks;
    }

    public void ReexamineRequest(int claim)
    {
        System.out.println("Reexamine request sent to student id " + id);
        controller.Notify(this, "reexamine", Integer.toString(claim));
    }
}
